### ISSUE TITLE
angularFire to persist the login when using custom generated JWT

### ISSUE THREAD

**Issue by exocom** [firebase/angularfire#189]
Thanks firebase team for making fireabase easy to use with angular!  I've recently switched from using the Email / Password auth and began using custom login on my angular app, however, I recently found out that I have to manage the persistence in order to keep the user signed in.

I really love the simplicity of using the Email / Pass and I have been able to build simple yet great angular apps really quick, however my needs have grown for several of my applications and I need to use custom generated tokens.

I would love to be able to leverage the simplicity of email login but pass the JWT generated from my server instead of the email / pass and have the angular app behave the same way.  Including supporting the the "remember me" flag for keeping the user signed in until the token expires.  This way I can leverage the great features of angularFireAuth such as route handling using the "authRequired" flag without having to rewrite it or add additional logic.

I feel having this baked into angularFire would raise the adoption rate of angularFire, and thus firebase, for developers who want to leverage firebase and angular but require custom tokens for added functionality and don't want to figure out auth inside of angular.   IE the same argument for creating angularFireAuth in the first place.

Basically since firebase has already written a great angular auth / persistence solution lets just extend it to include custom JWT!

Thanks,
Kalarrs
